src/core/modules/impl/handler/JsHandlerApi.ts

```diff
@@ -120,8 +120,14 @@ export class JsHandlerApi<State> extends AbstractContractHandler<State> {
       get: () => throwErrorWithName('ConstructorError', 'SmartWeave.block object is not accessible in constructor')
     });
 
+    const disabledTransaction = new Proxy(this.swGlobal.transaction, {
+      get: () =>
+        throwErrorWithName('ConstructorError', 'SmartWeave.transaction object is not accessible in constructor')
+    });
+
     this.swGlobal.vrf = disabledVrf;
     this.swGlobal.block = disabledBlock;
+    this.swGlobal.transaction = disabledTransaction;
```

Comment:
Disabled read comment below

Reply:
Left:

Warp.useCustom
Example in academy
refactor tags enum
I have also disabled access to transaction from constructor in this branch. Because I realised that it is not perfectly standardised and arlocal and gateway return diffrent transactions, which may lead to different results in test environment(local) vs mainnet.